### PR TITLE
Fixes this being able to absorb the same dna over and over as changeling.

### PR DESCRIPTION
**Pull request by Alberyk**
What it says in the title, now you can't steal someone's dna over and over as a changeling and just gets a lot of easy points.

### CODE REVIEW

**Comment by Kaedwuff** [comment]
Unless I'm mistaken, I already submitted a fix for this problem with #6492.  This change is going to sharply limit the ability for lings to function in a small crew round.

**Comment by Sindorman** [comment]
@Kaedwuff While your PR technically fixes it. It doesn't do same effect, this PR caps the max possible DNA a ling can achieve per round, whereas your PR keeps the max cap, but slows down them. If your PR is ready, please mark it as ready, as it has been sitting unlabled for a long time.

**Comment by Alberyk** [comment]
This pr adds real consequences and decision making for using those powers. If you use the stealthy sting, you won't be able to use the regular sting, that gives more points. Also, it stops people from taking the same person's dna over and over.

**Comment by Kaedwuff** [comment]
> This pr adds real consequences and decision making for using those powers. If you use the stealthy sting, you won't be able to use the regular sting, that gives more points. Also, it stops people from taking the same person's dna over and over.

The ability was supposed to give you more freedom as a ling to progress the round WITHOUT having to murder.  Making it a binary situation where you can only do one or another will cause a situation where lings are FORCED to kill on lowpop (and in fact, most rounds, because people have a tendency to avoid letting people near them once 'radiation warnings' start happening due to subtle meta) rounds or deadend themselves.

These 'real consequences' you speak of are little more than an enormous crippling nerf.

**Comment by OneOneThreeEight** [comment]
A guy stacked 26 genome points in 2 hours during one changeling round. He did not kill anyone.

This shouldn't be possible, sorry. I support this PR as-is.

**Comment by Alberyk** [comment]
> A guy stacked 26 genome points in 2 hours during one changeling round. He did not kill anyone.
> 
> This shouldn't be possible, sorry. I support this PR as-is.

image

I have seen the true face of this monster.

**Comment by Kaedwuff** [comment]
Leaping from no cooldown to something this extreme is intense knee-jerk.  This balanced be treated more carefully than with a mallet.  I understand you are all upset with the ability, but this is going to result in rounds where lings get less than ten DNA over the whole round, which is exactly what I was trying to avoid with this.

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 1]
```diff
@@ -175,6 +175,11 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
 		to_chat(src, "<span class='warning'>This creature's DNA is ruined beyond useability!</span>")
 		return
 
+	for(var/datum/absorbed_dna/D in changeling.absorbed_dna)
+		if(D.dna == T.dna)
+			to_chat(src, "<span class='warning'>We have already collected this creature's dna!</span>")
```
Comment (VTCobaltblood) [inline code/game/gamemodes/changeling/changeling_powers.dm]:
```suggestion
			to_chat(src, "<span class='warning'>We have already absorbed this creature's DNA!</span>")
```

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 2]
```diff
@@ -253,6 +258,11 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
 		to_chat(src, "<span class='warning'>We must have a tighter grip to absorb this creature.</span>")
 		return
 
+	for(var/datum/absorbed_dna/D in changeling.absorbed_dna)
+		if(D.dna == T.dna)
+			to_chat(src, "<span class='warning'>We have already collected this creature's dna!</span>")
```
Comment (VTCobaltblood) [inline code/game/gamemodes/changeling/changeling_powers.dm]:
```suggestion
			to_chat(src, "<span class='warning'>We have already absorbed this creature's DNA!</span>")
```

**Pull request merged**